script.js

```diff
@@ -54,83 +65,276 @@ function createEpisodeDescription(tvEpisodeDiv, episode) {
+
   allEpisodes.forEach((episode) => {
     const dropdownOption = document.createElement("option");
-    dropdownOption.value = `S${formatNumber(episode.season)}E${formatNumber(
-      episode.number
-    )}`;
-    dropdownOption.innerHTML = `S${formatNumber(episode.season)}E${formatNumber(
-      episode.number
-    )} - ${episode.name}`;
-    dropdownList.appendChild(dropdownOption);
+    dropdownOption.value = `${generateEpisodeCode(episode)}`;
+    dropdownOption.innerHTML = `${generateEpisodeCode(episode)} - ${
+      episode.name
+    }`;
+    episodeDropdownListEl.appendChild(dropdownOption);
   });
 
-  dropdownList.addEventListener("change", () => onEpisodeSelect());
+  episodeDropdownListEl.addEventListener("change", () => onEpisodeSelect());
 }
 
 // user makes a selection, they should be taken directly to that episode in the list
 function onEpisodeSelect() {
-  const selectedCode = document.getElementById("dropdownEpisode").value;
-
+  const selectedEpisodeId = document.getElementById("episodesDropdownList").value;
   const rootElem = document.getElementById("root");
 
-  const allEpisodes = getAllEpisodes();
-
-  const filteredEpisode = allEpisodes.filter((episode) => {
-      const episodeCode = `S${formatNumber(episode.season)}E${formatNumber(episode.number)}`;
-      return (episodeCode === selectedCode);
-    }
-  );
+  let filteredEpisode = [];
+  if (selectedEpisodeId === "showAll") {
+    filteredEpisode = allEpisodes;
+  } else {
+    filteredEpisode = allEpisodes.filter((episode) => {
+      const episodeCode = generateEpisodeCode(episode);
+      return episodeCode === selectedEpisodeId;
+    });
+  }
 
   rootElem.innerHTML = "";
   makePageForEpisodes(filteredEpisode);
 }
 
-window.onload = setup;
+//------------------------------ EPISODES API REQUEST ----------------------
+
+function getAllEpisodesFromApi() {
+  return fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => {
+      return response.json();
+    })
+    .then((episodes) => {
+      return episodes;
+    });
+}
+
+//------------------------------ SHOW DROPDOWN BAR ----------------------
+
+function showDropdownListOptions() {
+  const showDropdownListEl = document.getElementById("showsDropdownList");
+  const showAllOption = document.createElement("option");
+  showAllOption.value = "showAll";
+  showAllOption.innerHTML = "All Shows";
+  showDropdownListEl.appendChild(showAllOption);
+
+  allShows.forEach((show) => {
+    const dropdownOption = document.createElement("option");
+    dropdownOption.value = show.id;
+    dropdownOption.innerHTML = show.name;
+    showDropdownListEl.appendChild(dropdownOption);
+  });
+
+  showDropdownListEl.addEventListener("change", () => onShowSelect());
+}
+
+// user makes a selection, they should be taken directly to that show in the list
+function onShowSelect() {
+  const selectedShowId = document.getElementById("showsDropdownList").value;
+
+  if(selectedShowId === 'showAll') {
```

Comment:
And again, well done for handling showing all shows as an option in the dropdown